awx/main/migrations/0203_template_name_constraint.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 4.2.20 on 2025-04-22 15:54
+
+import logging
+
+from django.db import migrations, models
+
+from django.db.models import Count
+
+
+logger = logging.getLogger(__name__)
+
+
+def _rename_duplicates(cls):
+    for organization_id in cls.objects.order_by().values_list('organization_id', flat=True).distinct():
+        duplicate_data = cls.objects.values('name').filter(organization_id=organization_id).annotate(count=Count('name')).order_by().filter(count__gt=1)
+        for data in duplicate_data:
+            name = data['name']
+            ct = 0
+            for ujt in cls.objects.filter(name=name):
```

Comment:
we could considering only renaming the dups, not all of them. Maybe the most recent (by creation time) gets to keep the name, the rest get the suffix?

Reply:
Pushed a commit that implements this. Added some new testing to help get that worked out.